km/load_elf.c

```diff
@@ -93,26 +93,42 @@ static void load_extent(int fd, const Elf64_Phdr* phdr, km_gva_t base)
 
 static void km_find_dlopen(km_elf_t* e, km_gva_t adjust)
 {
-   if (e->symidx == -1 || e->stridx == -1) {
-      return;
+   int found = 0;
+   Elf64_Shdr sym_shdr;
+   Elf64_Shdr str_shdr;
+   for (int i = 0; i < e->ehdr.e_shnum; i++) {
+      if (km_elf_get_shdr(e, i, &sym_shdr) != 0) {
+         km_err(2, "Cannot get section header %d", i);
+      }
+
+      if (sym_shdr.sh_type == SHT_SYMTAB) {
+         if (km_elf_get_shdr(e, sym_shdr.sh_link, &str_shdr) != 0) {
+            km_err(2, "Cannot get section header %d", sym_shdr.sh_link);
+         }
+         found = 1;
+         break;
+      }
+   }
+   if (found == 0) {
+      km_err(2, "Cannot find symbol table for DLOPEN");
```

Comment:
table? Or just symbol?

Reply:
We are searching the section headers for the symbol table and it's associated string table. We look for the symbol itself further down.